File: src/main/java/co/herod/adbwrapper/util/PropHelper.java
package co.herod.adbwrapper.util;

import org.jetbrains.annotations.NotNull;

import java.util.Map;

@SuppressWarnings("WeakerAccess")
public class PropHelper {

    public static boolean hasPositiveValue(@NotNull Map.Entry<String, String> entry) {
        return hasPositiveValue(entry.getValue());
    }

    static boolean hasPositiveValue(String value) {

        value = value.trim().toLowerCase();

        return value.equals("on") || value.equals("1") || Boolean.parseBoolean(value);
    }

    public static boolean isValidProperty(@NotNull Map.Entry<String, String> a) {
        return !a.getKey().contains(" ");
    }

    public static boolean isKey(@NotNull Map.Entry<String, String> entry, String key) {
        return entry.getKey().equals(key);
    }
}

File: src/main/java/co/herod/adbwrapper/util/PropHelper.kt
package co.herod.adbwrapper.util

object PropHelper {

    fun hasPositiveValue(entry: Map.Entry<String, String>): Boolean {
        return entry.value.hasPositiveValue()
    }

    private fun String.hasPositiveValue(): Boolean {

        val value = trim { it <= ' ' }.toLowerCase()

        return value == "on" || value == "1" || value == "true"
    }

    fun isValidProperty(a: Map.Entry<String, String>): Boolean = a.key.contains(" ")

    fun isKey(entry: Map.Entry<String, String>, key: String): Boolean = entry.key == key

}
